coral/src/app/components/documentation/DocumentationEditor.tsx

```diff
@@ -0,0 +1,120 @@
+import { useState } from "react";
+import { DocumentationView } from "src/app/components/documentation/DocumentationView";
+
+type DocumentationEditorProps = {
+  documentation?: string;
+  save: (documentation: string) => void;
+  cancel: () => void;
+  isSaving?: boolean;
+};
+
+type ViewMode = "edit" | "preview";
+function DocumentationEditor({
+  documentation,
+  save,
+  cancel,
+  isSaving,
+}: DocumentationEditorProps) {
+  const [viewMode, setViewMode] = useState<ViewMode>("edit");
+  const [text, setText] = useState(documentation || "");
+
+  function saveDocumentation() {
+    // trim to remove unnecessary whitespace at end/start
+    const newDocumentation = text.trim();
+
+    // existing `documentation` needs to be trimmed to compare, since
+    // there we don't know if the value saved before migrating to React
+    // contains spaces at beginning / end
+    if (documentation && newDocumentation === documentation.trim()) return;
+    save(newDocumentation);
+  }
+
+  return (
+    <Box.Flex flexDirection={"column"} rowGap={"3"}>
+      <Box.Flex
+        alignSelf={"start"}
+        component={"section"}
+        aria-label={"Switch between edit and preview mode"}
+      >
+        <SegmentedControlGroup
+          onChange={(value: ViewMode) => setViewMode(value)}
+          value={viewMode}
+        >
+          <SegmentedControl aria-pressed={viewMode === "edit"} value={"edit"}>
+            Edit
+          </SegmentedControl>
+          <SegmentedControl
+            aria-pressed={viewMode === "preview"}
+            value={"preview"}
+          >
+            Preview
+          </SegmentedControl>
+        </SegmentedControlGroup>
+      </Box.Flex>
+
+      {viewMode === "edit" && (
+        <div className={classes.markdownEditor}>
+          <Box
+            borderColor={"grey-20"}
+            borderWidth={"1px"}
+            borderRadius={"2px"}
+            marginBottom={"2"}
+            className={classes.markdownSyntaxHighlight}
+          >
+            <label
+              className={"visually-hidden"}
+              htmlFor={"markdown-editor-textarea"}
+            >
+              Markdown editor
+            </label>
+            <textarea
+              id={"markdown-editor-textarea"}
+              aria-describedby={"editor-markdown-description"}
+              value={text}
+              onChange={(event) => setText(event?.target?.value)}
+              className={classes.markdownTextarea}
+            />
+            <div aria-hidden={"true"}>
+              <SyntaxHighlighter
+                language={"markdown"}
+                PreTag="div"
+                style={a11yLight}
+                wrapLongLines={true}
+                customStyle={{ minHeight: "35vh" }}
+              >
+                {text}
+              </SyntaxHighlighter>
+            </div>
+          </Box>
+          <Typography.SmallTextBold id={"editor-markdown-description"}>
```

Comment:
`SmallTextBold` is deprecated apparently :o Do you have support for it in your editor? :o
![Screenshot 2023-06-29 at 16 10 51](https://github.com/aiven/klaw/assets/20607294/12669f28-d84c-41f2-9165-dbe77f2e4889)

Reply:
Yes, I usually get that info 🤔  Updating!